packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -711,6 +870,38 @@ class ModalBottomSheetRoute<T> extends PopupRoute<T> {
 
     return capturedThemes?.wrap(bottomSheet) ?? bottomSheet;
   }
+
+  @override
+  Widget buildModalBarrier() {
+    Widget barrier;
+    if (barrierColor != null && barrierColor.alpha != 0 && !offstage) { // changedInternalState is called if barrierColor or offstage updates
+      assert(barrierColor != barrierColor.withOpacity(0.0));
+      final Animation<Color?> color = animation!.drive(
+        ColorTween(
+          begin: barrierColor.withOpacity(0.0),
+          end: barrierColor, // changedInternalState is called if barrierColor updates
+        ).chain(CurveTween(curve: barrierCurve)), // changedInternalState is called if barrierCurve updates
+      );
+      barrier = AnimatedModalBarrier(
```

Comment:
might as well directly return, same for the else case

Reply:
Thanks! Done.